apps/ios/SimpleXChat/ChatTypes.swift

```diff
@@ -3390,6 +3434,22 @@ public enum RcvGroupEvent: Decodable {
         case .groupUpdated: return NSLocalizedString("updated group profile", comment: "rcv group event chat item")
         case .invitedViaGroupLink: return NSLocalizedString("invited via your group link", comment: "rcv group event chat item")
         case .memberCreatedContact: return NSLocalizedString("connected directly", comment: "rcv group event chat item")
+        case let .memberProfileUpdated(fromProfile, toProfile): return profileUpdatedText(fromProfile, toProfile)
+        }
+    }
+
+    private func profileUpdatedText(_ from: Profile, _ to: Profile) -> String {
+        return if to.displayName != from.displayName || to.fullName != from.fullName {
+            String.localizedStringWithFormat(NSLocalizedString("member %@ changed to %@", comment: "rcv group event chat item"), from.profileViewName, to.profileViewName)
+        } else if to.image != from.image {
+            if to.image == nil {
+                profileUpdatedRemovedPictureText
+            } else {
+                profileUpdatedSetNewPictureText
+            }
+        } else {
+            // shouldn't happen if backend correctly creates item; UI should be synchronized with backend
+            defaultProfileUpdatedText
         }
     }
```

Comment:
```suggestion
    private func profileUpdatedText(_ from: Profile, _ to: Profile) -> LocalizedStringKey {
        to.displayName != from.displayName || to.fullName != from.fullName
        ? "member \(from.profileViewName) changed to \(to.profileViewName)"
        : to.image == from.image
        ? "updated profile" // won't happen
        : to.image == nil
        ? "removed profile picture"
        : "set new profile picture"
    }
```

No need for indirection, returns, ifs here.

Same for function above.

Reply:
It probably needs to be String as others are strings annoyingly, but it's easier to just repeat NSLocalizedString here with a comment than use constants, also easier to make them different if needed.

Reply:
ternary is harder to read as conditions and values are on the same level without indentation.

also this suggestion moves default case in the middle for optimization of one less comparison but makes it hard to read or change

Reply:
this more or less repeats logic in terminal view - check what's changed, then how
https://github.com/simplex-chat/simplex-chat/blob/master/src/Simplex/Chat/View.hs#L1429
except terminal view doesn't account for images and starts with default case